src/main/java/org/parabot/core/parsers/randoms/PublicRandoms.java

```diff
@@ -23,20 +23,29 @@
 
     @Override
     public void parse() {
-        File myJar = new File(Directories.getCachePath() + File.separator + fileName);
-        if (!myJar.exists() || !myJar.canRead()) {
-            download();
+
+        final File destination = new File(Directories.getCachePath() + File.separator + fileName);
+        final URL overrideDownload = Context.getInstance().getServerProviderInfo().getRandoms();
+        if (overrideDownload == null) {
+            throw new NullPointerException("randoms() URL is null - expected to default to BDN URL!");
```

Comment:
So you are required to fill in the randoms url?

Reply:
`   final URL overrideDownload = Context.getInstance().getServerProviderInfo().getRandoms();` will absolutely never be null, this just gives the NPE a message

Reply:
If it's absolutely never null, why would you add a nullcheck?

Reply:
I'll replace it with a comment. Shame we can't use @ NonNull :p

Reply:
Actually, I'll leave this exception in with an updated message. The URL will be null if the BDN URL ever changes, which of course it shouldn't, but there's no issue with some safety and a clarifying exception message.